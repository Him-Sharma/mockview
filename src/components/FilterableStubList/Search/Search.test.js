import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import Search from "./Search";

afterEach(cleanup);

it("renders correctly", () => {
  const { baseElement } = render(<Search />);
  expect(baseElement).toMatchSnapshot();
});
it("should notify when filter text is changed", () => {
  const mockCallback = jest.fn();
  const { getByTestId } = render(<Search onFilterTextChange={mockCallback} />);
  fireEvent.change(getByTestId("filterTextInput"), {
    target: { value: "test" },
  });

  expect(mockCallback).toHaveBeenCalledTimes(1);
  expect(mockCallback).toHaveBeenCalledWith("test");
});
