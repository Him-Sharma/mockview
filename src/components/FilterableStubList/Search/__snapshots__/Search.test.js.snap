// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <section
      class="search"
    >
      <div
        class="card"
      >
        <div
          class="search-input"
        >
          <label>
            Filter by Title
          </label>
          <input
            data-testid="filterTextInput"
            id="filterTextInput"
            type="text"
            value=""
          />
        </div>
      </div>
    </section>
  </div>
</body>
`;
